utils/uploads_used.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from datetime import datetime, timedelta
+
+from django.db.models import Q
+from django.utils import timezone
+
+from plan.constants import TrialStatus
+from reports.models import ReportSession
+from utils.config import get_config
+
+log = logging.getLogger(__name__)
+# default 6 hours for now
+cache_time = get_config("setup", "upload_usage_cache_time", 21600)
+
+
+def get_uploads_used(redis, plan_service, limit, owner):
+    cache_key = f"monthly_upload_usage_{owner.ownerid}"
+    try:
+        uploads_used = redis.get(cache_key)
+        if uploads_used is None:
+            uploads_used = query_uploads_used(plan_service, limit, owner)
+            redis.set(cache_key, uploads_used, ex=cache_time)
+        else:
+            uploads_used = int(uploads_used)
+
+    except OSError as e:
+        log.warning(
+            f"Error connecting to redis for rate limit check: {e}",
+            extra=dict(owner=owner.ownerid),
+        )
+        uploads_used = query_uploads_used(plan_service, limit, owner)
+    return uploads_used
+
+
+def set_uploads_used(redis, owner, uploads_used):
```

Comment:
[nit] this function seems to be unused for now, could you just leave a comment for what you foresee it would be used for